public/app/plugins/datasource/grafana-azure-monitor-datasource/grafanaTemplateVariableFns.ts

```diff
@@ -295,6 +298,7 @@ const migrateGrafanaTemplateVariableFn = (query: AzureMonitorQuery) => {
       break;
   }
 
+  delete migratedQuery.grafanaTemplateVariableFn;
```

Comment:
not sure if this was left on purpose (maybe to track migrations or roll back?)

If no reason then I think removing it makes the panel JSON cleaner

Reply:
Removing it breaks almost every test in `variables.test.ts` with timeout errors 🤔 That's why I added it back, do you have any idea why that is?

Reply:
those tests use this method:

```
      observables.subscribe((result: DataQueryResponseData) => {
        expect(result.data[0].source).toEqual(expectedResults);
        done();
      });
```

if the `expect` fails, `done` is never called so the test hangs until reaching the timeout. If you scroll, you should be able to see the real error.

Reply:
Yes, I get a `Cannot read properties of undefined (reading 'source')` meaning `result.data[0]` is undefined there. 
I tried logging the result but got nothing back 🤔 

I don't see how that could relate to removing the `grafanaTemplateVariableFn` property 🤷‍♀️ . 

Should I merge without the `delete` for a quick fix? Or you think it could mean there is another bug hidden somewhere that the delete surfaces?

Reply:
I haven't checked the code but that probably means that  `grafanaTemplateVariableFn` is used elsewhere (either in functional code or in the test). But since there is no need to remove the property I would merge this as it is.